app/src/main/java/org/piwigo/bg/AlbumService.java

```diff
@@ -0,0 +1,69 @@
+package org.piwigo.bg;
+
+import android.accounts.Account;
+import android.app.IntentService;
+import android.content.Intent;
+import android.widget.Toast;
+
+import org.piwigo.io.RestService;
+import org.piwigo.io.RestServiceFactory;
+import org.piwigo.io.model.AddCategoryResponse;
+
+import javax.inject.Inject;
+
+import dagger.android.AndroidInjection;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class AlbumService extends IntentService {
+
+    public static final String KEY_CATEGORY_NAME = "category_name";
+    public static final String KEY_PARENT_CATEGORY_ID = "category_id";
+    public static final String KEY_ACCOUNT = "account";
+
+    @Inject
+    RestServiceFactory restServiceFactory;
+
+    public AlbumService() {
+        super("AlbumService");
+    }
+
+    @Override
+    public void onCreate() {
+        AndroidInjection.inject(this);
+        super.onCreate();
+    }
+
+    @Override
+    protected void onHandleIntent(Intent intent) {
+        String catName = intent.getStringExtra(KEY_CATEGORY_NAME);
+        Account curAccount = intent.getParcelableExtra(KEY_ACCOUNT);
+        int parentId = intent.getIntExtra(KEY_PARENT_CATEGORY_ID, 0);
+
+        RestService restService = restServiceFactory.createForAccount(curAccount);
+
+        Call<AddCategoryResponse> call = restService.addCategory(catName, parentId, null, null, null, null);
+
+        call.enqueue(new Callback<AddCategoryResponse>() {
+            @Override
+            public void onResponse(Call<AddCategoryResponse> call, Response<AddCategoryResponse> response) {
+                if (response.raw().code() == 200) {
+                    if (response.body().stat.equals("ok")) {
+                        // TODO: make text localizable
```

Comment:
Shouldn't we do the 'todo' before we merge it into master?

Reply:
A bunch of things are waiting for localization; I guess we could do all of them in a row.

Reply:
we could, but we could also do it for those things here directly...

I don't see a benefit in making this to-be-planned effort bigger and I think it will get quite boring to do all the localization stuff in one step. - You know that you'd not need to translate it yourself, but we use crowdin to do that?
And in general it would be better to not add a TODO but in case we really want to defer some task create a ticket for it and put the ticket number there (maybe with TODO), to not loose track over what is to be done... Otherwise we will also make the "one task to fix ALL TODOs" just bigger.

Reply:
and btw which are the places where we need localization? Shouldn't we create a ticket for them?